packages/formdescriptioneditors/backend/sirius-components-collaborative-formdescriptioneditors/src/main/java/org/eclipse/sirius/components/collaborative/formdescriptioneditors/handlers/AddWidgetEventHandler.java

```diff
@@ -94,23 +98,35 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
         changeDescriptionSink.tryEmitNext(changeDescription);
     }
 
-    protected boolean addWidget(IEditingContext editingContext, IFormDescriptionEditorContext formDescriptionEditorContext, String kind, int index) {
-        var optionalSelf = this.objectService.getObject(editingContext, formDescriptionEditorContext.getFormDescriptionEditor().getTargetObjectId());
+    protected boolean addWidget(IEditingContext editingContext, IFormDescriptionEditorContext formDescriptionEditorContext, String containerId, String kind, int index) {
```

Comment:
why `protected`?

Reply:
So this method was protected in order to write a test without mocking the object service properly. By writing the test properly, it appears that the test could not really work in normal circumstances and the code below could fail with a bug too.